### DISCUSSION TITLE
Path aliases fails to resolve "@/*"

### DISCUSSION THREAD

**Discussion by dinesh-hills** [hngngn/shadcn-solid#55] [Q&A] [answered]
`[plugin:vite:import-analysis] Failed to resolve import "@/libs/cn" from "src/components/ui/button.tsx". Does the file exist?`

Despite I have done everything correct.
It works when using relative path `../../libs/cn`
Why is it the problem with solid vite plugin?

I am using plain solidjs with Typescript

**Comment by hngngn** [comment, marked as answer]
you need to update your path in `tsconfig.json` to "@"
```ts
{
  "compilerOptions": {
    "baseUrl": "./",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
```

 and add this to your `vite.config.ts` 
```ts
resolve: {
  alias: {
    "@": resolve(__dirname, "./src")
  }
}
```

you can read how to setup [manual](https://shadcn-solid.com/docs/installation/manual) too.

**Reply by dinesh-hills** [reply]
Thanks for the quick response, Yes I did exactly as per the manual, I missed the vite config part which was not mentioned in the manual.

**Reply by hngngn** [reply]
I'm adding new installation guide for astro too so i will add this in the next update.
